wallet-new/test/unit/Test/Spec/Addresses.hs

```diff
@@ -123,6 +126,38 @@ prepareAddressFixture n = do
                 let SliceOf{..} = Addresses.getAddresses (RequestParams pp) db'
                 return . map AddressFixture $ paginatedSlice
 
+prepareAddressesFixture
+    :: Int  -- ^ Number of Accounts to create.
+    -> Int  -- ^ Number of 'Address per account to create.
+    -> Fixture.GenPassiveWalletFixture (M.Map V1.AccountIndex [V1.WalletAddress])
+prepareAddressesFixture acn adn = do
+    spendingPassword <- Fixture.genSpendingPassword
+    newWalletRq <- WalletLayer.CreateWallet <$> Wallets.genNewWalletRq spendingPassword
+    return $ \pw -> do
+        let newAcc (n :: Int) = (V1.NewAccount spendingPassword ("My Account " <> show n))
+        Right v1Wallet <- Wallets.createWallet pw newWalletRq
+        forM_ [1..acn] $ \n ->
+            Accounts.createAccount pw (V1.walId v1Wallet) (newAcc n)
+        -- Get all the available accounts
+        db <- Kernel.getWalletSnapshot pw
+        let Right accs = Accounts.getAccounts (V1.walId v1Wallet) db
+        let accounts = IxSet.toList accs
+        length accounts `shouldBe` (acn + 1)
+        let insertAddresses :: V1.Account -> IO (V1.AccountIndex, [V1.WalletAddress])
+            insertAddresses acc = do
+                let accId = V1.accIndex acc
+                let newAddressRq = V1.NewAddress spendingPassword accId (V1.walId v1Wallet)
+                res <- replicateM adn (Addresses.createAddress pw newAddressRq)
+                case sequence res of
+                    Left e     -> error (show e)
+                    Right addr -> return (accId, addr)
+        res <- mapM insertAddresses accounts
+        return $ M.fromList res
+
+expectedNumber :: Int -> Int -> Int
+expectedNumber acc adr = (acc + 1)*(adr + 1) - acc
```

Comment:
I think if you bake the logic inside the fixture, you might be able to remove this.